onnxruntime/python/tools/transformers/fusion_rotary_attention.py

```diff
@@ -578,7 +714,18 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
         self.node_name_to_graph_name[new_node.name] = self.this_graph_name
 
         self.nodes_to_remove.extend(qkv_nodes[1:])
-        self.nodes_to_remove.extend(v_nodes[:-1])
+
+        if v_nodes != v_nodes_4:
+            self.nodes_to_remove.extend(v_nodes[:-1])
+        else:
+            remove_dic = {}
+            node_keep_name = v_nodes[0][-1].name
+            for temp_path in v_nodes:
+                for temp_node in temp_path:
+                    if temp_node.name not in remove_dic and temp_node.name != node_keep_name:
```

Comment:
Can we remove some of the condition checks by using this function instead?
https://github.com/microsoft/onnxruntime/blob/785e2b1eaec2b2e69267cc9320f7749315c0e686/onnxruntime/python/tools/transformers/fusion_base.py#L117-L132

Reply:
Seems not easy to use that function directly but I can extract these logic into a separate function and put into fusion_base